common/internal_model/src/main/scala/uk/ac/wellcome/models/parse/FreeformDate.scala

```diff
@@ -0,0 +1,103 @@
+package uk.ac.wellcome.models.parse
+
+import java.time.LocalDate
+
+import uk.ac.wellcome.models.work.internal.InstantRange
+
+/**
+  *  Models freeform text representing some date or period
+  *
+  *  @param date The date or date range
+  *  @param inferred Whether the date is known exactly or inferred
+  */
+case class FreeformDate(date: Either[FuzzyDate, DateRange],
+                        inferred: Boolean = false) {
+
+  def instantRange: Option[InstantRange] =
+    (date match {
+      case Left(date)       => date.instantRange
+      case Right(dateRange) => dateRange.instantRange
+    }) map { _ withInferred inferred }
+
+  def withInferred(inferred: Boolean): FreeformDate =
+    FreeformDate(date, inferred)
+}
+
+/**
+  *  A exact or ambigous date
+  */
+sealed trait FuzzyDate extends DateHelpers {
+
+  def instantRange: Option[InstantRange] =
+    this match {
+
+      case ExactDate(date) =>
+        Some(InstantRange(date))
+
+      case Year(year) =>
+        Some(InstantRange(yearStart(year), yearEnd(year)))
+
+      case MonthAndYear(month, year) =>
+        Some(InstantRange(monthStart(month, year), monthEnd(month, year)))
+
+      case _ => None
+    }
+}
+
+case class ExactDate(date: LocalDate) extends FuzzyDate
+case class Year(year: Int) extends FuzzyDate
+case class MonthAndYear(month: Int, year: Int) extends FuzzyDate
+case class Month(day: Int) extends FuzzyDate
+case class Day(day: Int) extends FuzzyDate
+
+/**
+  *  A continuos period over some days / months / years
+  *
+  *  @param from The start date
+  *  @param to The end date
+  */
+case class DateRange(from: FuzzyDate, to: FuzzyDate) extends DateHelpers {
```

Comment:
I wonder if calling this a `FuzzyDateRange` might have helped with drawing the relationships between these types?

Reply:
Yes played around a lot with the name of these thing but I think that could be clearer, cheers :+1

Reply:
Have changed to `FuzzyDateRange` now :)